### ISSUE TITLE
Unable to Access Login Page While Logged in as Unauthorized User

### ISSUE THREAD

**Issue by tobiasiv** [payloadcms/payload#10180]
### Describe the Bug

Currently, the login page cannot be accessed if you are logged in as a non-admin-panel-user.
For example, consider a scenario where you have two collections: Users and Customers. The following configuration is set:

```tsx
const config = buildConfig({
  // ...
  admin: { 
    user: Users.slug,
  },
  // ...
})
```
This configuration allows only users from the Users collection to login to the admin panel.

If you are logged in as a customer and try to access the login page, Payload detects an active session from the Customers collection. It attempts to redirect you to the admin panel, but since customers are not authorized to access the panel, the following error message appears:

"Unauthorized, you must be logged in to make this request."

This message is misleading. It should only appear when a user of the Users collection lacks the necessary permissions.

However, if you are logged in as a customer, you should still be able to access the login page to switch accounts and log in as an admin-panel-user (Not simultaneously, but by switching auth).

### Link to the code that reproduces this issue

https://github.com/tobiasiv/payload-reproduction/tree/unauthorized

### Reproduction Steps

1. Clone the reproduction repo (branch: [unauthorized](https://github.com/tobiasiv/payload-reproduction/tree/unauthorized))
2. Set the env variables
3. `pnpm dev`
4. Open http://localhost:3000 and http://localhost:3000/admin in the browser
5. Press the Login Button of the Admin Login Page
6. Press the Login Customer Button of the Frontend
7. Refresh the Admin Dashboard
8. An error appears: "Unauthorized, you must be logged in to make this request"
9. To access the admin panel login page again you have to press the Logout Customer Button of the Frontend

### Which area(s) are affected? (Select all that apply)

area: core

### Environment Info

```text
Binaries:
  Node: 20.16.0
  npm: N/A
  Yarn: N/A
  pnpm: 9.7.0
Relevant Packages:
  payload: 3.11.0
  next: 15.1.0
  @payloadcms/db-postgres: 3.11.0
  @payloadcms/email-nodemailer: 3.11.0
  @payloadcms/graphql: 3.11.0
  @payloadcms/next/utilities: 3.11.0
  @payloadcms/payload-cloud: 3.11.0
  @payloadcms/richtext-lexical: 3.11.0
  @payloadcms/ui/shared: 3.11.0
  react: 19.0.0
  react-dom: 19.0.0
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home
  Available memory (MB): 16263
  Available CPU cores: 8
```

**Comment by jacobsfletch** [comment]
Hey @tobiasiv I've got a fix for this queued up in #10188. Instead of allowing public users through to the login page, we instead needed to properly allow them to _log out_ through the provided link. This was currently broken as that method was using an incorrect API route. I've also improved the messaging around this.

**Comment by tobiasiv** [comment]
> Hey [@tobiasiv](https://github.com/tobiasiv) I've got a fix for this queued up in [#10188](https://github.com/payloadcms/payload/pull/10188). Instead of allowing public users through to the login page, we instead needed to properly allow them to _log out_ through the provided link. This was currently broken as that method was using an incorrect API route. I've also improved the messaging around this.

Hey @jacobsfletch ,
Thanks for the fix!

Since the login page is accessible to any public user, wouldn’t it make sense to allow non-admin-panel users to log in as well? Only admin-panel users (defined in payload.config.ts) should be redirected to the admin panel and gain access based on their permissions.

For collections like Customers with auth enabled, they should be treated as public users. If they attempt to access the admin panel, they should remain on the login page unless they log in as an admin-panel user. (This could change in the future if multiple simultaneous auth become possible)

Additionally, if a customer logs into the admin panel, their session should switch — logging them out as a customer and logging them in as an admin-panel user.

Thanks again for the quick response and ongoing improvements!
